src/dataset.py

```diff
@@ -0,0 +1,68 @@
+from pathlib import Path
+
+import lightning as L
+import rasterio as rio
+import torch
+from torch.utils.data import DataLoader, Dataset
+
+
+class ClayDataset(Dataset):
+    def __init__(self, data_dir: Path, transform=None):
+        super().__init__()
+        self.data_dir = data_dir
+        self.chips_path = list(data_dir.glob("**/*.tif"))
+        self.transform = transform
+
+    def read_chip(self, chip_path):
+        ts = chip_path.parent.name
+        year, month, day = map(int, ts.split("-"))
+
+        chip = rio.open(chip_path)
+        bounds = chip.bounds
+        centroid_x = (bounds.left + bounds.right) / 2
+        centroid_y = (bounds.top + bounds.bottom) / 2
+
+        return {
+            "pixels": chip.read(),
+            "timestep": (year, month, day),
+            "latlon": (centroid_x, centroid_y),
```

Comment:
A tuple like (centroid_x, centroid_y) would be lonlat, not latlon. Longitude is X, latitude in Y.

Reply:
Thanks @weiji14 swapping the values